Project2015To2017/ProjectConverter.cs

```diff
@@ -189,6 +188,7 @@ private IReadOnlyCollection<ITransformation> TransformationsToApply(bool modernP
 				new AssemblyAttributeTransformation(this.logger, this.conversionOptions.KeepAssemblyInfo),
 				new XamlPagesTransformation(this.logger),
 				new PrimaryUnconditionalPropertyTransformation(),
+				new PropertySimplificationTransformation(),
```

Comment:
No, you can't move `PropertySimplificationTransformation` here.
* `PropertyDeduplicationTransformation` depends on simplification
* `PrimaryUnconditionalPropertyTransformation` depends on `PropertyDeduplicationTransformation`

So no way this can be moved this late. If you want to have properties, generated by `AssemblyAttributeTransformation` simplified — better move `AssemblyAttributeTransformation` to be before `PropertySimplificationTransformation`.

Reply:
I remember now, the thing that meant I had to move it after the `PrimaryUnconditionalPropertyTransformation` was this bit in there:

```xml
			var propertyGroup = project.PrimaryPropertyGroup();
			propertyGroup.Add(project.AssemblyAttributeProperties);
```

Until that point, the `project.AssemblyAttributeProperties` haven't made it into the main property group, so they aren't available to be simplified. 

A few other options I thought of here:
1. Get rid of `AssemblyAttributeProperties` and just chuck them all in to PrimaryPropertyGroup. That would affect a lot of tests and might make users with custom transforms have to work harder (I think that includes me as I do custom things with the versions).
2. Make it a derived property that just fetches any of the associated properties from `project.PrimaryPropertyGroup` after they have been put there. I'd need to handle defaults here too though.
3. I could add them to the `project.PrimaryPropertyGroup` in the AssemblyAttributeTransform rather than the one above (or as well as and rely on the deduplication transform).

I think I will go for 3 as the simplest for now

Reply:
Yeah, 3rd is the best one, but 2nd can in addition to 3rd be implemented not to break code relying on presence of `AssemblyAttributeProperties`.
But the second part is terrible, don't rely on deduplication for that — that transformation is for conditional `PropertyGroup`s only.

Reply:
Ok, I've now just changed it to do it in `AssemblyAttributeTransform`. There might be scope to do something better in future to reduce the duplication but it's no worse than it was.